api/types.go

```diff
@@ -183,8 +178,15 @@ type Options struct {
 	EmbeddingOnly      bool    `json:"embedding_only,omitempty"`
 	RopeFrequencyBase  float32 `json:"rope_frequency_base,omitempty"`
 	RopeFrequencyScale float32 `json:"rope_frequency_scale,omitempty"`
+	NumThread          int     `json:"num_thread,omitempty"`
+}
 
-	// Predict options
+type Options struct {
+	Runner
```

Comment:
Instead of splitting Options and embedding Runner, you can create functions `Options.RuntimeOptions` and `Options.RunnerOptions` to present different views of `Options`

Reply:
I leaned to embedding Runner just so we didnt have to worry about adding fields in two places in the future, where you add the field makes it obvious if it effects loading the model